c2corg_ui/static/js/search.js

```diff
@@ -163,44 +172,55 @@ app.SearchController = function($scope, $compile, $attrs, apiUrl, gettextCatalog
 
 /**
  * @param {string} type The document type.
- * @param {string} title The category title.
  * @return {TypeaheadDataset} A data set.
  * @private
  */
-app.SearchController.prototype.createDataset_ = function(type, title) {
+app.SearchController.prototype.createDataset_ = function(type) {
   var bloodhoundEngine = this.createAndInitBloodhound_(type);
   return /** @type {TypeaheadDataset} */({
+    contents : type,
     source: bloodhoundEngine.ttAdapter(),
     display: function(doc) {
       return doc.label;
     },
     limit: 20,
     templates: {
       header: (function() {
-        return '<div class="header ' + title + '">' +
-            this.gettextCatalog_.getString(title) + '</div>';
+        return '<div class="header" dataset="' + type + '">' + this.gettextCatalog_.getString(type) + '</div>';
       }).bind(this),
+      footer: function() {
+        return '<p class="suggestion-more"><a href="/' + type + '" class="green-text">+ see more results</a></p>'
+      },
       suggestion: function(doc) {
-        return '<p>' + doc.label + '</p>';
-      }
-      // Note: The templates for `notFound` and `pending` are not set, because
-      // they would be shown for each dataset.
+        if (doc.documentType === 'routes') {
+          return this.compile_('<suggestion query="' + doc['_query'] + '" label="' + doc['label'] + '" elevation="' + doc['elevation_max']  + '" activities="' + doc['activities'] + '" doctype="routes" class="tt-suggestion"></suggestion>')(this.scope_);
+        } else if (doc.documentType === 'waypoints') {
+          return this.compile_('<suggestion query="' + doc['_query'] + '" label="' + doc['label'] + '" elevation="' + doc['elevation'] + '" type="' + doc['waypoint_type'] + '" doctype="waypoints" class="tt-suggestion"></suggestion>')(this.scope_);
+        } else if (doc.documentType === 'outings') {
+          return this.compile_('<suggestion query="' + doc['_query'] + '" label="' + doc['label']  + '" elevation="' + doc['elevation_max'] + '" activities="' + doc['activities'] + '" doctype="outings" class="tt-suggestion"></suggestion>')(this.scope_);
+        }
+      }.bind(this),
+      empty: function(res) {
+        if ($('.header.empty').length === 0) {
+          var datasetId = res.dataset;
+          var dataset = this.datasets[datasetId];
+          return this.createEmptyResults_(dataset.contents);
+        }
+      }.bind(this)
     }
   });
 };
 
-
 /**
  * @param {string} type The document type.
  * @return {Bloodhound} The bloodhound engine.
  * @private
  */
 app.SearchController.prototype.createAndInitBloodhound_ = function(type) {
-  var empty_template = $('#empty-results').html(); // in base.html
   var url = this.apiUrl_ + '/search?q=%QUERY';
 
   var bloodhound = new Bloodhound(/** @type {BloodhoundOptions} */({
-    limit: 10,
+    limit: 7,
```

Comment:
it took too much space in my opinion, like 10 results took more than half of the screen height.